tests/test_dimension.py

```diff
@@ -1,21 +0,0 @@
-import devito
-import pytest
-from conftest import skipif_yask
-from devito import Grid, FixedDimension, Function, y
-
-
-@skipif_yask
-@pytest.mark.xfail
-def test_incorrect_usage():
-    grid = Grid(shape=(10, 10))
-    m = Function(name="m", grid=grid)
-    devito.x = FixedDimension(name='x', size=10, spacing=devito.x.spacing)
-    assert(devito.x in m.indices)
-
-
-@skipif_yask
-def test_correct_usage():
-    myx = FixedDimension(name='x', size=10, spacing=devito.x.spacing)
-    grid = Grid(shape=(10, 10), dimensions=(myx, y))
-    m = Function(name="m", grid=grid)
-    assert(myx in m.indices)
```

Comment:
Maybe not for this PR, but should we still have at least one test (no idea what) for Dimensions?

Reply:
We have a whole bunch of tests that use explicitly specified and grid-derived dimensions all over the test base - this test was really just specific to the fixed dimension API and it's recent change. I do agree though that we are lacking tests for some more exotic dimension combinations, in particular with respect to mixing space/time and auxiliary dimensions - but yes that is definitely stuff for future PRs.

Reply:
> do agree though that we are lacking tests for some more exotic dimension combinations,

We made some with @fablup previously as part of some dse fixes. I think there is some left in #340 that include mixed dimensions laplacian. I can dig it up, and can be part of a new PR

Reply:
Excellent. I'm still working on adding dedicated `TimeDimension`  types, so these will come in handy.

Reply:
#340 is actually quite small, the `test_operator` with `b.laplace` would the go to one for this PR to make sure we catch correct dimensions for auto-shortcuts